lib/components/viewers/trip-viewer.js

```diff
@@ -87,89 +102,94 @@ class TripViewer extends Component {
-                    {highlightClass && <div className={highlightClass} />}
-                    <div className={stripMapLineClass} />
-                    <div className="strip-map-icon">
-                      <StyledIconWrapper>
-                        <Circle />
-                      </StyledIconWrapper>
+          <StopList>
+            {tripData &&
+              tripData.stops &&
+              tripData.stopTimes &&
+              tripData.stops.map((stop, i) => {
+                // determine whether to use special styling for first/last stop
+                let stripMapLineClass = 'strip-map-line'
+                if (i === 0) stripMapLineClass = 'strip-map-line-first'
+                else if (i === tripData.stops.length - 1) {
+                  stripMapLineClass = 'strip-map-line-last'
+                }
+
+                // determine whether to show highlight in strip map
+                let highlightClass
+                if (i === viewedTrip.fromIndex) {
+                  highlightClass = 'strip-map-highlight-first'
+                } else if (i > viewedTrip.fromIndex && i < viewedTrip.toIndex) {
+                  highlightClass = 'strip-map-highlight'
+                } else if (i === viewedTrip.toIndex) {
+                  highlightClass = 'strip-map-highlight-last'
+                }
+
+                return (
+                  <Stop key={i}>
+                    {/* the departure time */}
+                    <div className="stop-time">
+                      <DepartureTime
+                        originDate={startOfDay}
+                        stopTime={tripData.stopTimes[i]}
+                      />
                     </div>
-                  </div>
-
-                  {/* the stop-viewer button */}
-                  <div className="stop-button-container">
-                    <ViewStopButton
-                      stopId={stop.id}
-                      text={
-                        <FormattedMessage id="components.TripViewer.viewStop" />
-                      }
-                    />
-                  </div>
-
-                  {/* the main stop label */}
-                  <div className="stop-name">{stop.name}</div>
-
-                  <div style={{ clear: 'both' }} />
-                </div>
-              )
-            })}
+
+                    {/* the vertical strip map */}
+                    <div className="strip-map-container">
+                      {highlightClass && <div className={highlightClass} />}
+                      <div className={stripMapLineClass} />
+                      <div className="strip-map-icon">
+                        <StyledIconWrapper>
+                          <Circle />
+                        </StyledIconWrapper>
+                      </div>
+                    </div>
+
+                    {/* the stop-viewer button */}
+                    <div className="stop-button-container">
+                      <ViewStopButton
```

Comment:
Note (TODO): To a screen reader, "View" appears before the stop name. A separate label for screen readers will need to be set up.